tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -174,4 +176,297 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+	*/
+	ginkgo.It("[csi-block-vanilla][csi-block-vanilla-parallelized] Fill LZT/EZT volume", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sharedvmfsURL := os.Getenv(envSharedVMFSDatastoreURL)
+		if sharedvmfsURL == "" {
+			ginkgo.Skip(fmt.Sprintf("Env %v is missing", envSharedVMFSDatastoreURL))
+		}
+
+		largeSize := os.Getenv(envDiskSizeLarge)
+		if sharedvmfsURL == "" {
+			largeSize = diskSizeLarge
+		}
+
+		scParameters := make(map[string]string)
+		policyNames := []string{}
+		pvcs := []*v1.PersistentVolumeClaim{}
+		pvclaims2d := [][]*v1.PersistentVolumeClaim{}
+		scs := []*storagev1.StorageClass{}
+
+		rand.Seed(time.Now().UnixNano())
+		suffix := fmt.Sprintf("-%v-%v", time.Now().UnixNano(), rand.Intn(10000))
+		categoryName := "category" + suffix
+		tagName := "tag" + suffix
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		allocationTypes := []string{
+			eztAllocType,
+			lztAllocType,
+		}
+
+		ginkgo.By("create SPBM policies with LZT, EZT volume allocation respectively")
+		ginkgo.By("Create SCs using policies created in step 1")
+		ginkgo.By("create a large PVC each using the storage policies created from step 2")
+		for _, at := range allocationTypes {
+			policyID, policyName := createVmfsStoragePolicy(
+				ctx, pc, at, map[string]string{categoryName: tagName})
+			defer func() {
+				deleteStoragePolicy(ctx, pc, policyID)
+			}()
+			scParameters[scParamStoragePolicyName] = policyName
+			policyNames = append(policyNames, policyName)
+			storageclass, pvclaim, err := createPVCAndStorageClass(client,
+				namespace, nil, scParameters, largeSize, nil, "", false, "")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pvcs = append(pvcs, pvclaim)
+			pvclaims2d = append(pvclaims2d, []*v1.PersistentVolumeClaim{pvclaim})
+			scs = append(scs, storageclass)
+		}
+
+		defer func() {
+			ginkgo.By("Delete the SCs created in step 2")
+			for _, sc := range scs {
+				err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		for i, pv := range pvs {
+			volumeID := pv.Spec.CSI.VolumeHandle
+			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
+			e2eVSphere.verifyVolumeCompliance(volumeID, true)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+		}
+
```

Comment:
Can we check here if volume has been created on the expected datastore?

Reply:
done